test/test-actionlib.js

```diff
@@ -0,0 +1,341 @@
+// Copyright (c) 2017 Intel Corporation. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* eslint-disable camelcase */
+
+'use strict';
+
+const assert = require('assert');
+const rclnodejs = require('../index.js');
+
+describe('Testing message files generated from an action file', function() {
+
+  this.timeout(60 * 1000);
+
+  beforeEach(function() {
+    return rclnodejs.init();    
+  });
+
+  afterEach(function() {
+    rclnodejs.shutdown();
+  });
+
+  it('ActionServer start', function() {
+    let as = new rclnodejs.ActionLib.ActionServer({
+      type: 'ros1_actions/msg/DoDishes',
+      actionServer: 'dishes1',
+      rclnodejs: rclnodejs
+    });
+
+    as.start();
+  });
+
+  it('GoalStatus requiring', function() {
+    rclnodejs.require('actionlib_msgs/msg/GoalStatus');
+  });
+
+  it('ActionClient sendGoal', function() {
+    let ac = new rclnodejs.ActionClientInterface({
+      type: 'ros1_actions/msg/DoDishes',
+      actionServer: 'dishes3',
+      rclnodejs: rclnodejs
+    });
+
+    ac.sendGoal({ goal: {dishwasher_id: 3}});
+  });
+
+  it('ActionLib server goal event triggered', function(done) {
+    let as = new rclnodejs.ActionLib.ActionServer({
+      type: 'ros1_actions/msg/DoDishes',
+      actionServer: 'dishes4',
+      rclnodejs: rclnodejs
+    });
+    as.on('goal', function(goal) {
+      done();
+    });
+    as.start();
+
+    let ac = new rclnodejs.ActionClientInterface({
+      type: 'ros1_actions/msg/DoDishes',
+      actionServer: 'dishes4',
+      rclnodejs: rclnodejs
+    });
+    ac.sendGoal({ goal: {dishwasher_id: 4}});
```

Comment:
I saw the time out happened on CI when running this case, Could you try to execute `sendGoal` periodically OR set an interval to avoid the contention.

Reply:
I'll take a try.